apps/jobs/models.py

```diff
@@ -119,7 +119,12 @@ def save(self, *args, **kwargs):
                 participant_team=self.participant_team,
                 status=Submission.FAILED).count()
 
-            successful_count = self.submission_number - failed_count
+            cancelled_count = Submission.objects.filter(
```

Comment:
Instead of making a fresh query, you can do something like this:

```
            submissions = Submission.objects.filter(
                challenge_phase=self.challenge_phase,
                participant_team=self.participant_team)
           submission_status_to_exclude = [Submission.CANCELLED, Submission.FAILED]
           num_submissions_to_ignore = submissions.filter(status__in=submission_status_to_exclude)
```

Note that you have to remove the logic on line 117-121.